src/zenml/models/v2/core/model_version_artifact.py

```diff
@@ -0,0 +1,178 @@
+#  Copyright (c) ZenML GmbH 2023. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Models representing the link between model versions and artifacts."""
+
+from typing import TYPE_CHECKING, Any, Dict, Optional, Union
+from uuid import UUID
+
+from pydantic import Field, validator
+
+from zenml.models.v2.base.base import (
+    BaseResponse,
+    BaseResponseBody,
+    BaseResponseMetadata,
+)
+from zenml.models.v2.base.scoped import (
+    WorkspaceScopedFilter,
+    WorkspaceScopedRequest,
+)
+
+if TYPE_CHECKING:
+    from zenml.models.v2.core.artifact_version import ArtifactVersionResponse
+
+
+# ------------------ Request Model ------------------
+
+
+class ModelVersionArtifactRequest(WorkspaceScopedRequest):
+    """Request model for links between model versions and artifacts."""
+
+    model: UUID
+    model_version: UUID
+    artifact_version: UUID
+    is_model_artifact: bool = False
+    is_endpoint_artifact: bool = False
```

Comment:
What is an endpoint artifact?

Reply:
An endpoint artifact was and will be a deployment artifact

Reply:
Hope that helps